pandas/core/arrays/datetimes.py

```diff
@@ -2232,7 +2232,7 @@ def maybe_convert_dtype(data, copy: bool, tz: tzinfo | None = None):
     if is_float_dtype(data.dtype):
         # pre-2.0 we treated these as wall-times, inconsistent with ints
         # GH#23675, GH#45573 deprecated to treat symmetrically with integer dtypes
-        data = data.astype(np.int64)
+        data = data.astype("M8[ns]").view("i8")
```

Comment:
if this works, can you add a comment about why its needed

Reply:
seems to have worked in https://app.circleci.com/pipelines/github/pandas-dev/pandas/18643/workflows/978d3b67-4eb0-4d53-b42d-081554172b25/jobs/49433

sure, have added a comment